src/components/portrait/portrait.component.js

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import tagComponent from '../../utils/helpers/tags';
+import OptionsHelper from '../../utils/helpers/options-helper';
+import StyledPortrait, {
+  StyledPortraitInitials, StyledPortraitGravatar, StyledCustomImg, StyledIcon
+} from './portrait.style';
+
+function Portrait(props) {
+  const {
+    alt, darkBackground, gravatar, initials, shape, size, src
+  } = props;
+
+  const showIcon = Boolean(!src && !initials);
+  const showInitials = Boolean(!src && initials);
+  const showGravatar = Boolean(gravatar);
+  const showCustomImg = Boolean(!gravatar && src);
+
+  return (
+    <StyledPortrait
+      showIcon={ showIcon }
+      showInitials={ showInitials }
+      showGravatar={ showGravatar }
+      showCustomImg={ showCustomImg }
+      { ...props }
+      { ...tagComponent('portrait', props) }
+    >
+
+      {showIcon
```

Comment:
I think it should be good to extract this to separate function and in this place just return the function. I think It would be more readable